content/300-guides/050-database/830-prismas-type-system.mdx

```diff
@@ -0,0 +1,152 @@
+---
+title: How to use Prisma's type system
+metaDescription: How to use Prisma's type system
+tocDepth: 2
+---
+
+<TopBlock>
+
+This guide introduces Prisma's type system and explains how to introspect existing native types in your database, and how to use types when you apply schema changes to your database with Prisma Migrate or `db push`.
+
+</TopBlock>
+
+## How does Prisma's type system work?
+
+Prisma uses _types_ to define the kind of data that a field can hold. To make it easy to get started, Prisma provides a small number of core [scalar types](/reference/api-reference/prisma-schema-reference#model-field-scalar-types) that should cover most default use cases. For example, in the blog post model below, the `title` field uses the `String` scalar type, while the `createdAt` field uses the `DateTime` scalar type:
+
+```prisma file=schema.prisma
+model Post {
+  id        Int      @id
+  title     String
+  createdAt DateTime
+}
+```
+
+Databases also have their own type system, which defines the type of value that a column can hold. Most databases provide a large number of data types to allow fine-grained control over exactly what a column can store. For example, a database might provide inbuilt support for multiple sizes of integers, or for XML data. The names of these types vary between databases. In PostgreSQL a variable length text column has a type of `text`, whereas in MySQL the type is `varchar`.
```

Comment:
The example in the last sentence is less clear cut than it could be, because postgres also has `VARCHAR` and MySQL also has `TEXT`. One alternative example could be the column type for booleans: `BOOLEAN` on Postgres, but  `TINYINT(1)` is what people typically use on MySQL.

Reply:
Thanks, that's definitely a better example. I've changed it to that.